src/serial-console/azext_serialconsole/custom.py

```diff
@@ -695,3 +671,92 @@ def enable_serialconsole(cmd):
 def disable_serialconsole(cmd):
     client = cf_serialconsole(cmd.cli_ctx)
     return client.disable_console()
+
+
+def get_region_from_storage_account(cli_ctx, resource_group_name, vm_vmss_name, vmss_instanceid):
+    from azext_serialconsole._client_factory import storage_client_factory
+
+    result = None
+    storage_account_region = None
+    client = _compute_client_factory(cli_ctx)
+    scf = storage_client_factory(cli_ctx)
+
+    if vmss_instanceid:
+        result_data = client.virtual_machine_scale_set_vms.get_instance_view(
+            resource_group_name, vm_vmss_name, vmss_instanceid)
+        result = result_data
+
+        if result_data.boot_diagnostics is None:
+            error_message = "Azure Serial Console requires boot diagnostics to be enabled."
+            recommendation = ('Use "az vmss update --name MyScaleSet --resource-group MyResourceGroup --set '
+                              'virtualMachineProfile.diagnosticsProfile="{\\"bootDiagnostics\\": {\\"Enabled\\" : '
+                              '\\"True\\",\\"StorageUri\\" : null}}"" to enable boot diagnostics. '
+                              'You can replace "null" with a custom storage account '
+                              '\\"https://mystor.blob.windows.net/"\\. Then run "az vmss update-instances -n '
+                              'MyScaleSet -g MyResourceGroup --instance-ids *".')
+            raise AzureConnectionError(
+                error_message, recommendation=recommendation)
+        else:
+            if result.boot_diagnostics is not None:
+                print(result.boot_diagnostics)
```

Comment:
Is this `print` for debug? Do you want to keep it in release version?

Reply:
That print was for debug purposes and I would like to keep it in the release version if possible.  Is there a different way to implement that for debug purposes?

Reply:
Please use `logger.debug()` then. This would be print only when `--debug` parameter specified

Reply:
Please import logger from knack:
```
from knack.log import get_logger
logger = get_logger(__name__)
```

Reply:
I added the missing import and kicked off the build again